### DISCUSSION TITLE
Does velero not support EKS roles for Service Accounts?

### DISCUSSION THREAD

**Discussion by ghost** [vmware-tanzu/velero#3954] [Community support Q&A]
I could only get velero working with IAM user and id/secret. I tried to use the "role" method (similar to how it's done with kube2iam); but slightly adjusted to create my own role (with proper policy permissions) and just passed it via the service account but kept getting permission denied errors from velero pod.

Details on what I'm talking about [is here](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)

Basically:
1. Create a role and attach it to the cluster via namespace/user
2. Deploy the pod/deployment in the namespace with the service account

This method is how I deploy autoscaler, external-dns, etc; so it not working weakens security footprint of our cluster (we only use roles and no kube2iam).

**Comment by ghost** [comment]
I was able to get it working after following this post that uses IAM Role for Service Account:

https://www.fourco.nl/blogs/backup-and-restore-a-kubernetes-cluster-with-velero/

**Reply by drewhemm** [reply]
When deploying with Helm, I found it necessary to add the following values:

```yaml
credentials:
  useSecret: false
```
